wled00/FX_fcn.cpp

```diff
@@ -1511,15 +1521,28 @@ void WS2812FX::loadCustomPalettes()
 
       if (readObjectFromFile(fileName, nullptr, &pDoc)) {
         JsonArray pal = pDoc[F("palette")];
-        if (!pal.isNull() && pal.size()>7) { // not an empty palette (at least 2 entries)
-          size_t palSize = MIN(pal.size(), 72);
-          palSize -= palSize % 4; // make sure size is multiple of 4
-          for (size_t i=0; i<palSize && pal[i].as<int>()<256; i+=4) {
-            tcp[ i ] = (uint8_t) pal[ i ].as<int>(); // index
-            tcp[i+1] = (uint8_t) pal[i+1].as<int>(); // R
-            tcp[i+2] = (uint8_t) pal[i+2].as<int>(); // G
-            tcp[i+3] = (uint8_t) pal[i+3].as<int>(); // B
-            DEBUG_PRINTF("%d(%d) : %d %d %d\n", i, int(tcp[i]), int(tcp[i+1]), int(tcp[i+2]), int(tcp[i+3]));
+        if (!pal.isNull() && pal.size()>4) { // not an empty palette (at least 2 entries)
+          if (pal[0].is<int>() && pal[1].is<const char *>()) {
+            // we have an array of index & hex strings
+            size_t palSize = MIN(pal.size(), 36);
+            palSize -= palSize % 2; // make sure size is multiple of 2
+            for (size_t i=0, j=0; i<palSize && pal[i].as<int>()<256; i+=2, j+=4) {
+              uint8_t rgbw[] = {0,0,0,0};
+              tcp[ j ] = (uint8_t) pal[ i ].as<int>(); // index
+              colorFromHexString(rgbw, pal[i+1].as<const char *>()); // will catch non-string entires
+              for (size_t c=0; c<3; c++) tcp[j+1+c] = rgbw[c]; // only use RGB component
+              DEBUG_PRINTF("%d(%d) : %d %d %d\n", i, int(tcp[j]), int(tcp[j+1]), int(tcp[j+2]), int(tcp[j+3]));
```

Comment:
Maybe comment out these debug_print once it works ... because people will not understand what that means when running a debug build.